source/AI.cpp

```diff
@@ -2760,6 +2769,41 @@ void AI::DoScatter(Ship &ship, Command &command)
 
 
 
+bool AI::DoSecretive(Ship &ship, Command &command)
+{
+	shared_ptr<Ship> scanningShip;
+	// Figure out if any ship is currently scanning us. If that is the case, move away from it.
+	for(auto &otherShip : GetShipsList(ship, false))
+		if(!ship.GetGovernment()->Trusts(otherShip->GetGovernment()) &&
+				otherShip->Commands().Has(Command::SCAN) &&
+				otherShip->GetTargetShip() == ship.shared_from_this() &&
+				!otherShip->IsDisabled() && !otherShip->IsDestroyed())
+			scanningShip = make_shared<Ship>(*otherShip);
+
+	if(scanningShip)
+	{
+		Point scanningPos = scanningShip->Position();
+		Point pos = ship.Position();
+
+		double cargoDistanceSqrd = 10000. * scanningShip->Attributes().Get("cargo scan power");
+		double outfitDistanceSqrd = 10000. * scanningShip->Attributes().Get("outfit scan power");
+
+		double maxScanRangeSqrd = max(cargoDistanceSqrd, outfitDistanceSqrd);
+		double distanceSqrd = scanningPos.DistanceSquared(pos);
```

Comment:
This is how Ship::Scan does it.
```suggestion
		double cargoDistanceSqrd = scanningShip->Attributes().Get("cargo scan power");
		double outfitDistanceSqrd = scanningShip->Attributes().Get("outfit scan power");

		double maxScanRangeSqrd = max(cargoDistanceSqrd, outfitDistanceSqrd);
		double distanceSqrd = scanningPos.DistanceSquared(pos) * .0001;
```

Reply:
applied locally